### PR TITLE
Add export to allow external scripts to get killer data

### PR DESCRIPTION
**Pull request by JHansen2000**
## Description

Add export to allow external scripts to get killer data

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

**Comment by JHansen2000** [comment]
Sorry for the messy commits, I had some old code I was playing around with that was fixed by someone else's changes and I forgot to remove that old code prior to opening this PR.

**Review by Manason — CHANGES_REQUESTED** [review]
Would this be better as an event instead?

**Comment by mafewtm** [comment]
What is the usecase for this as well?

**Comment by JHansen2000** [comment]
This is helpful for developers to be able to retrieve the killer's data for other scripts (i.e. a deathscreen that displays killer name and how they killed you). Without this export, external scripts must listen to the native CEventNetworkEntityDamage and parse it on their own, which is very inefficient since Qbox is doing this already.

I think this is better as an export, rather than an event, but am definitely open to discussion if you think an event would be better. What is the benefit of using an event over an export for this?

**Comment by JHansen2000** [comment]
> > This is helpful for developers to be able to retrieve the killer's data for other scripts (i.e. a deathscreen that displays killer name and how they killed you). Without this export, external scripts must listen to the native CEventNetworkEntityDamage and parse it on their own, which is very inefficient since Qbox is doing this already.
> 
> > 
> 
> > I think this is better as an export, rather than an event, but am definitely open to discussion if you think an event would be better. What is the benefit of using an event over an export for this?
> 
> 
> 
> How would this be used? The event makes sense to me so that other scripts can listen for an onDeath event if they care to take action when the player dies. Otherwise, they will have to call the export in a loop to continuously check.

I see where you're coming from. My scripts listen for changes to the client state bag instead of an event, so I only call the export when the state changes. I think this is good practice, but let me know if you disagree.

**Comment by Manason** [comment]
> > > This is helpful for developers to be able to retrieve the killer's data for other scripts (i.e. a deathscreen that displays killer name and how they killed you). Without this export, external scripts must listen to the native CEventNetworkEntityDamage and parse it on their own, which is very inefficient since Qbox is doing this already.
> > 
> > 
> > > 
> > 
> > 
> > > I think this is better as an export, rather than an event, but am definitely open to discussion if you think an event would be better. What is the benefit of using an event over an export for this?
> > 
> > 
> > How would this be used? The event makes sense to me so that other scripts can listen for an onDeath event if they care to take action when the player dies. Otherwise, they will have to call the export in a loop to continuously check.
> 
> I see where you're coming from. My scripts listen for changes to the client state bag instead of an event, so I only call the export when the state changes. I think this is good practice, but let me know if you disagree.

Wouldn't it be more streamlined to have a named event with additional data (such as the killer data) rather than state bag listener + export?

**Comment by JHansen2000** [comment]
It would be more streamlined. I will update this tonight

**Comment by JHansen2000** [comment]
Sorry for the delay, this has been added simply as an event trigger. I do think this (and other events/exports) should be added to documentation somewhere but did not want to modify the README without approval.

File: client/dead.lua [hunk 1]
```diff
@@ -125,6 +125,7 @@ AddEventHandler('gameEventTriggered', function(event, data)
         DeathTime = config.deathTime
         OnDeath()
     end
+    TriggerEvent('qbx_medical:client:attackerData', attacker, weapon)
```
Comment (Manason) [inline client/dead.lua:128]: Pass this data to the OnDeath function and to the existing onPlayerDied event. No need for a new event. https://github.com/Qbox-project/qbx_medical/blob/380465254d4af3f1c8c6d6e4c2f7ea0c3645647f/client/dead.lua#L26

**Review by Manason — CHANGES_REQUESTED** [review]

**Comment by JHansen2000** [comment]
I passed the data as props to both OnDeath and StartLastStand and from there into the associated client & server events. It's all optional but is definitely very helpful. Hope that looks okay and this wasn't too much of a headache for you!

**Review by Manason — APPROVED** [review]

**Pull request merged**